packages/@aws-cdk/aws-cloudwatch/test/test.metrics.ts

```diff
@@ -124,4 +142,82 @@ export = {
 
     test.done();
   },
+
+  'throws error when there are more than 10 dimensions in dimensionsMap'(test: Test) {
+    test.throws(() => {
+      new Metric({
+        namespace: 'Test',
+        metricName: 'ACount',
+        period: cdk.Duration.minutes(10),
+        dimensionsMap: {
+          dimensionA: 'value1',
+          dimensionB: 'value2',
+          dimensionC: 'value3',
+          dimensionD: 'value4',
+          dimensionE: 'value5',
+          dimensionF: 'value6',
+          dimensionG: 'value7',
+          dimensionH: 'value8',
+          dimensionI: 'value9',
+          dimensionJ: 'value10',
+          dimensionK: 'value11',
+        },
+      } );
+    }, /The maximum number of dimensions is 10, received 11/);
+
+    test.done();
+  },
+
+  'can make metric with dimensionsMap property'(test: Test) {
+    const m = new Metric({
+      namespace: 'Test',
+      metricName: 'Metric',
+      period: cdk.Duration.minutes(10),
+      dimensionsMap: {
+        dimensionA: 'value1',
+        dimensionB: 'value2',
+      },
+    });
+
+    expect(m).toMatch({
```

Comment:
This is a weird assertion... We really should be using something like `expect(stack).to(haveResourceLike(...` instead. I don't really care what are the exact properties of `Metric` here, just that it generates the correct CloudFormation at the end.

It might be valuable to have an assertion on `metric` (I would avoid naming a variable `m`) `.dimensions` though.

Reply:
Since there is no CloudFormation resource for Metric (it is only a property on an AWS::Cloudwatch::Alarm or inside the DashboardBody of AWS::Cloudwatch::Dashboard), I can't do a stack level assertion without also creating an alarm or a dashboard. But, it would be pretty simple to just also create an Alarm for this test. Thoughts?

Reply:
And are you saying to replace this `toMatch()` with only checking the value of `metric.dimensions`?

Reply:
I would create the simplest possible resource that allows me to write a `expect(stack).to(haveResourceLike(...` assertion that confirms the rendered dimensions are what we expect, and also have an assertions like `test.deepEqual(metrics.dimensions, blah)`.

Reply:
Okay makes sense, thanks! Updated the PR with that.